src/main/java/org/springframework/data/elasticsearch/core/mapping/SimpleElasticsearchPersistentEntity.java

```
@@ -385,6 +400,36 @@ private EvaluationContext getIndexNameEvaluationContext() {
 		return getEvaluationContext(null, expressionDependencies);
 	}
 
+	@Override
+	public String resolveRouting(Object bean) {
+
+		if (routing == null) {
+			return null;
+		}
+
+		ElasticsearchPersistentProperty persistentProperty = getPersistentProperty(routing);
+
+		if (persistentProperty != null) {
+			Object propertyValue = getPropertyAccessor(bean).getProperty(persistentProperty);
+
+			return propertyValue != null ? propertyValue.toString() : null;
+		}
+
+		try {
+			Expression expression = routingExpressions.computeIfAbsent(routing, PARSER::parseExpression);
+			ExpressionDependencies expressionDependencies = ExpressionDependencies.discover(expression);
+
+			EvaluationContext context = getEvaluationContext(null, expressionDependencies);
+			context.setVariable("entity", bean);
+
+			return expression.getValue(context, String.class);
+		} catch (EvaluationException e) {
+			LOGGER.warn("could not resolve", e);
```

How about throwing an exception instead of silently falling back to `null`?